src/WebJobs.Script/Workers/Profiles/HostPropertyCondition.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Workers.Profiles;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers
+{
+    internal class HostPropertyCondition : IWorkerProfileCondition
+    {
+        private readonly ILogger _logger;
+        private readonly ISystemRuntimeInformation _systemRuntimeInformation;
+        private readonly string _name;
+        private readonly string _expression;
+        private Regex _regex;
+
+        public HostPropertyCondition(ILogger logger, ISystemRuntimeInformation systemRuntimeInformation, WorkerProfileConditionDescriptor descriptor)
+        {
+            if (descriptor is null)
+            {
+                throw new ArgumentNullException(nameof(descriptor));
+            }
+
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            _systemRuntimeInformation = systemRuntimeInformation ?? throw new ArgumentNullException(nameof(systemRuntimeInformation));
+
+            descriptor.Properties.TryGetValue(WorkerConstants.WorkerDescriptionProfileConditionName, out _name);
+            descriptor.Properties.TryGetValue(WorkerConstants.WorkerDescriptionProfileConditionExpression, out _expression);
+
+            Validate();
+        }
+
+        public enum ConditionHostPropertyName
+        {
+            Sku,
+            Platform,
+            HostVersion
+        }
+
+        public string Name => _name;
+
+        public string Expression => _expression;
+
+        public bool Evaluate()
+        {
+            string value = string.Empty;
+            Enum.TryParse(Name, out ConditionHostPropertyName hostPropertyName);
+            switch (hostPropertyName)
```

Comment:
supernit: adding a space here so it's cleaner to read
```suggestion

            switch (hostPropertyName)
```